components/support/nimbus-fml/src/backends/swift/templates/EnumTemplate.swift

```diff
@@ -0,0 +1,17 @@
+{% import "macros.swift" as swift %}
+{% let inner = self.inner() %}
+{% let class_name = inner.name()|class_name %}
+{{ inner.doc()|comment("") }}
+public enum {{ class_name }}: String {
+    {% for variant in inner.variants() %}
+    {{ variant.doc()|comment("    ") }}
+    case {{ variant.name()|enum_variant_name }} = {{variant.name()|quoted}}
+    {% endfor %}
+
+    public static func enumValue(_ s: String?) -> {{class_name}}? {
+        if let s = s {
+            return {{class_name}}(rawValue: s)
+        }
+        return nil
```

Comment:
`if let` is a `nil` check, so it's redundant here.

```suggestion
        return {{class_name}}(rawValue: s)
```

Reply:
Yeah, the nil check is intentional here because the `enumValue` in swift takes a `String?`, since the `v.getString("the-property")` returns a `String?`